### DISCUSSION TITLE
Automatic shutdown

### DISCUSSION THREAD

**Discussion by DrRSatzteil** [sle118/squeezelite-esp32#250] [Q&A] [answered]
I was wondering if anyone has implemented an automatic shutdown feature with squeezelite.

I'd like to build a small battery powered music player with an esp32 audio kit. I'd like to be able to switch the box on and off with a simple push button which should be doable with a latching circuit that keeps the power running while the esp is on or the power is plugged in.

However I would be interested to power off the device when it has not played any sound for a given time (basically just change the state of a GPIO after e.g. 10 minutes without sound). I implemented this behaviour on raspi device before but I don't know if it would be possible to do something like this with squeezelite. Does anyone have any ideas how this could be done?

**Comment by sle118** [comment, marked as answer]
You should have a look at the frame readme. This is probably already supported.

https://github.com/sle118/squeezelite-esp32#set-gpio

**Reply by DrRSatzteil** [reply]
True... I read over it so many times but somehow still missed it. Thank you!

**Comment by sle118** [comment]
@DrRSatzteil you should have a look again. @philippe44 implemented a sleep feature, which sends the system to sleep when there's no playback for a certain period of time.
